ec2/networkinterface.py

```diff
@@ -0,0 +1,213 @@
+    }
+    # If this is an external resource, set the ID and exit
+    if params.get('use_external_resource'):
+        _update_runtime_properties(
+            _get_network_interface_object(
+                ec2_client, params.get('resource_id')))
+        return
+
+    # Find a Subnet if an ID wasn't explicitly provided
+    if 'subnet_id' not in params:
+        subnet = _find_attached_subnet()
+        if not subnet:
+            raise NonRecoverableError(
+                'Network Interface without a Subnet is not allowed')
+        params['subnet_id'] = \
+            subnet.instance.runtime_properties.get('aws_resource_id')
+    # Find attached Security Groups
+    params['groups'] = list(set(
+        _find_attached_security_groups() + params.get('groups', list())))
+    # Create the Network Interface
+    ctx.logger.debug('create_network_interface({0})'.format(params))
+    try:
+        iface_obj = ec2_client.create_network_interface(**params)
+        # Create tags if needed
+        if isinstance(ctx.node.properties.get('tags'), dict):
+            ec2_client.create_tags([iface_obj.id], ctx.node.properties['tags'])
+            # Update the object
+            iface_obj = _get_network_interface_object(ec2_client, iface_obj.id)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError) as ex:
+        raise NonRecoverableError('{0}'.format(str(ex)))
+    # Update the runtime properties
+    _update_runtime_properties(iface_obj)
+
+
+@operation
+def delete(**_):
+    '''Deletes an Elastic Network Interface'''
+    if not ctx.node.properties.get('use_external_resource') and \
+       ctx.instance.runtime_properties.get('aws_resource_id'):
+        ec2_client = connection.EC2ConnectionClient().client()
+        try:
+            # Delete the Network Interface
+            ec2_client.delete_network_interface(
+                ctx.instance.runtime_properties.get('aws_resource_id'))
+        except (boto.exception.EC2ResponseError,
+                boto.exception.BotoServerError) as ex:
+            raise NonRecoverableError('{0}'.format(str(ex)))
+
+
+@operation
+def attach_elastic_ip(**_):
+    '''Attaches an Elastic IP to this Elastic Network Interface'''
+    ec2_client = connection.EC2ConnectionClient().client()
+    # Get the Elastic Netowrk Interface runtime properties
+    eni_props = ctx.source.instance.runtime_properties
+    # Attempt to associate the EIP with the ENI
+    try:
+        _get_address_object(
+            ec2_client,
+            ctx.target.instance.runtime_properties.get('aws_resource_id')
+        ).associate(
+            network_interface_id=eni_props.get('aws_resource_id'),
+            private_ip_address=eni_props.get('aws_private_ip_address'))
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError) as ex:
+        raise NonRecoverableError('{0}'.format(str(ex)))
+
+
+@operation
+def detach_elastic_ip(**_):
```

Comment:
use the [existing code](https://github.com/cloudify-cosmo/cloudify-aws-plugin/blob/master/ec2/elasticip.py#L176). if there are functional....